### PR TITLE
feat: add trailing stops, special instructions, and high-value feature parity gaps

### PR DESCRIPTION
**Pull request by major**
## Summary

Close the remaining high-value feature parity gaps: trailing stop orders, special instructions, quote field filtering, MOC/LOC order types, first-triggers-second (FTS) composite orders, order repeat command, and Order-to-OrderRequest conversion.

### Order types and builders
- **Trailing stop orders**: `TRAILING_STOP` and `TRAILING_STOP_LIMIT` equity order support with `--stop-offset` flag and validation rules
- **MOC/LOC order types**: `MARKET_ON_CLOSE` and `LIMIT_ON_CLOSE` aliases ("MOC"/"LOC") in `parseOrderType`, validation in `ValidateEquityOrder`, price field handling in `applyEquityPriceFields`
- **FTS builder**: `FTSParams` struct, `ValidateFTSOrder`/`BuildFTSOrder` functions, and `order build fts --primary --secondary` CLI command
- **Special instructions**: `--special-instruction` flag for equity and option orders (ALL_OR_NONE, DO_NOT_REDUCE, ALL_OR_NONE_DO_NOT_REDUCE)

### New commands
- **`order repeat <id>`**: Fetch an existing order and re-submit it with `--build` (default, output JSON), `--preview` (dry-run via API), or `--confirm` (place order). Safety guards enforced before any API calls.
- **`order build fts`**: Build first-triggers-second composite orders from JSON specs

### Quote enhancements
- **`QuoteParams`**: New struct with `Fields []string` and `Indicative bool` for filtering API responses
- **`--fields` and `--indicative` flags**: Added to `quote get` with case-insensitive validation and comma-separated field support

### Conversion utility
- **`OrderToRequest()`**: Explicit allowlist field copy from `Order` (API response) to `OrderRequest` (API submission) with recursive `ChildOrderStrategies` handling

### Tests
- 60+ new test cases across all features: table-driven subtests, httptest mock servers, typed error assertions, edge case coverage
- All existing tests continue to pass

### Docs
- FEATURE_PARITY.md updated: all 8 high-value gaps marked as done

### CODE REVIEW

File: internal/models/order.go [hunk 1]
```diff
@@ -559,3 +559,43 @@ type PreviewOrder struct {
...
+// Only fields that exist in OrderRequest are copied from Order.
+// ChildOrderStrategies are recursively converted from []Order to []OrderRequest.
+// Response-only fields (OrderID, Status, FilledQuantity, etc.) are intentionally excluded.
+func OrderToRequest(order *Order) OrderRequest {
+	req := OrderRequest{
+		Session:                  order.Session,
+		Duration:                 order.Duration,
+		OrderType:                order.OrderType,
+		ComplexOrderStrategyType: order.ComplexOrderStrategyType,
+		Quantity:                 order.Quantity,
```
[the remark this answers is not in the document]
Comment (major) [reply internal/models/order.go:580]: Fixed. CancelTime is now in the allowlist, with test coverage in both models_test.go and order_test.go (sampleOrderJSON includes it, assertions verify round-trip).

File: internal/commands/order.go [hunk 2]
```diff
@@ -386,6 +399,98 @@ func orderReplaceCommand(c *client.Ref, configPath string, w io.Writer) *cli.Com
...
+			// Enforce safety guards early for --confirm so we fail fast
+			// before making any API calls.
+			if confirmMode {
+				if err := requireMutableEnabled(configPath); err != nil {
+					return err
+				}
+				if err := requireConfirm(cmd.Bool("confirm")); err != nil {
+					return err
+				}
+			}
```
[the remark this answers is not in the document]
Comment (major) [reply internal/commands/order.go:443]: Fixed. Removed the tautological `requireConfirm` call and added a comment explaining that `--confirm` doubles as both mode selector and safety gate.

File: internal/models/order.go [hunk 3]
```diff
@@ -559,3 +559,43 @@ type PreviewOrder struct {
...
+
+	// Recursively convert child order strategies from []Order to []OrderRequest.
+	if len(order.ChildOrderStrategies) > 0 {
+		req.ChildOrderStrategies = make([]OrderRequest, len(order.ChildOrderStrategies))
+		for i := range order.ChildOrderStrategies {
+			req.ChildOrderStrategies[i] = OrderToRequest(&order.ChildOrderStrategies[i])
+		}
+	}
+
+	return req
```
[the remark this answers is not in the document]
Comment (major) [reply internal/models/order.go:607]: Fixed. CancelTime added to the OrderToRequest allowlist with dedicated test coverage (`cancel time is preserved` subtest in models_test.go, `cancelTime` field in sampleOrderJSON with assertion in order_test.go).

File: internal/models/order.go [hunk 4]
```diff
@@ -559,3 +559,43 @@ type PreviewOrder struct {
...
+		StopPriceLinkBasis:       order.StopPriceLinkBasis,
+		StopPriceLinkType:        order.StopPriceLinkType,
+		StopPriceOffset:          order.StopPriceOffset,
+		StopType:                 order.StopType,
+		PriceLinkBasis:           order.PriceLinkBasis,
+		PriceLinkType:            order.PriceLinkType,
+		PriceOffset:              order.PriceOffset,
+		Price:                    order.Price,
+		TaxLotMethod:             order.TaxLotMethod,
+		OrderLegCollection:       order.OrderLegCollection,
```
[the remark this answers is not in the document]
Comment (major) [reply internal/models/order.go]: Fixed. OrderLegCollection now gets a defensive copy via `append([]OrderLegCollection(nil), ...)` so the request and original order don't share a backing array.

File: internal/orderbuilder/complex.go [hunk 5]
```diff
@@ -216,6 +216,32 @@ func bracketExitInstruction(action models.Instruction) models.Instruction {
...
+	if err := ValidateFTSOrder(params); err != nil {
+		return models.OrderRequest{}, err
+	}
+
+	params.Primary.OrderStrategyType = models.OrderStrategyTypeTrigger
+	params.Secondary.OrderStrategyType = models.OrderStrategyTypeSingle
+	params.Primary.ChildOrderStrategies = []models.OrderRequest{params.Secondary}
+
+	return params.Primary, nil
+}
```
[the remark this answers is not in the document]
Comment (major) [reply internal/orderbuilder/complex.go:247]: Fixed. BuildFTSOrder now copies params by value into local variables before mutating, consistent with BuildBracketOrder/BuildOCOOrder.
